### ISSUE TITLE
Latest update crashes XrmToolBox

### ISSUE THREAD

**Issue by peterloudon** [Innofactor/XrmToolBox.PluginTraceViewer#65]
After updating to the latest version of PluginTraceViewer, XrmToolBox issues a message to say it will gather statistics for PluginTraceViewer and then crashes. This is repeated whenever further updates are attempted, but not if updates are bypassed. If the plugin is removed from XrmToolBox, everything else works. If all other plugins are removed, this works until a second plugin is installed (FetchXmlBuilder), after which the errors return. XrmToolBox log is attached. Error has been reproduced on two separate machines, on different domains.
[XrmToolBox Error.txt](https://github.com/Innofactor/XrmToolBox.PluginTraceViewer/files/1331665/XrmToolBox.Error.txt)

**Comment by rappen** [comment]
Hi - sorry to hear that. The log did not reveal the exact reason for the crash to me, but as a general suggestion you should try to remove all files matching:
```
c:\Users\<yourusername>\AppData\Roaming\MscrmTools\XrmToolBox\Plugins\Cinteros.*
c:\Users\<yourusername>\AppData\Roaming\MscrmTools\XrmToolBox\Plugins\Innofactor.*
```
Then start XrmToolBox and reinstall the plugins you want.
It happens that different versions of plugins reference dependent assemblies, and there is a version mismatch. The reinstall of the plugins *should* solve that problem.

**Comment by peterloudon** [comment]
Hi, Thanks for the feedback. I have tried the suggestion, but with no change. If you review my original message, you will see that I have actually gone so far as to remove all plugins and install (and update) only PluginTraceViewer plus one other (XrmToolBox). I have also installed (and updated) all plugins except for PluginTraceViewer. Either way, I get the error. What information would assist in troubleshooting this?

**Comment by rappen** [comment]
I'm actually at a loss here. Perhaps @mscrmtools can help?

**Comment by MscrmTools** [comment]
@peterloudon XrmToolBox log contains an error when preparing updates from Plugins Store. When I read your explanations, it seems the error occurs when running Plugin Trace Viewer...
Can you confirm when exactly the error occurs?

**Comment by peterloudon** [comment]
@MscrmTools, the item referenced by @rappen is the same as mine.

XrmToolBox was running fine until the latest update for PluginTraceViewer was installed. After it was installed, the problem arose on updates. As long as I try no further updates, XrmToolBox will run. As soon as I try to update, the error appears.

To try to remove, or at least isolate, the problem, I did the following, on two different, unrelated Windows 10 environments:

1. Completely removed XrmToolBox (including the whole MscrmTools folder structure in AppData).
2. Downloaded the latest copy of XrmToolBox, unlocked it, and installed it.
3. Uninstalled all plugins.
4. Installed and updated PluginTraceViewer.
5. Installed and updated FetchXmlBuilder, AND got the error.

1. Completely removed XrmToolBox (including the whole MscrmTools folder structure in AppData).
2. Downloaded the latest copy of XrmToolBox, unlocked it, and installed it.
3. Uninstalled PluginTraceViewer.
4. Installed and updated all plugins other than PluginTraceViewer.
5. Installed and updated PluginTraceViewer, AND got the error.

The final error which is displayed before exiting is attached below. It suggests a problem with a modified collection and an enumeration operation which may not execute. (The suggestion that files are locked is not applicable - see step 2 above).

I am not sure if this is related, but I am attaching a screen grab of my updates window. The five updates listed will not install. This started before the problem with PluginTraceViewer. When I select them and click Install, the application shuts down and reopens, but the updates are still offered. (I have not installed XTBPlugin because it is the tutorial application and should probably not be there.)

Thanks both for your assistance.


![xrmtoolbox error](https://user-images.githubusercontent.com/16725000/31062995-9e405d34-a78b-11e7-8ece-e6ee4f0f859e.png)
![xrmtoolbox updates](https://user-images.githubusercontent.com/16725000/31063050-30902c28-a78c-11e7-9034-2b045f45b4d1.png)

**Comment by rappen** [comment]
Fixed in commit https://github.com/Innofactor/XrmToolBox.PluginTraceViewer/commit/914f2411568ae2cb340a334ddd91be23daa4b19c

**Comment by rappen** [comment]
Fix is now published in version 1.2017.10.1.
